### ISSUE TITLE
runtime: go test inconsistent line number (by runtime.Caller) with and w/o -race

### ISSUE THREAD

**Issue by h12w** [golang/go#13249]
## go version

go version go1.5.1 linux/amd64
## what I did
1. paste the following code to any_test.go (with no preceeding empty lines)
2. go test
3. go test -race

``` go
package any

import (
    "runtime"
    "testing"
)

func TestGetPos(t *testing.T) {
    line := someFunc( // this is the line that should be restored.
        0,
        1,
        2,
    )
    if expected := 9; line != expected {
        t.Fatalf(`Expect "%v" to be "%v"`, line, expected)
    }
}

func someFunc(int, int, int) int {
    return GetPos(1)
}

// GetPos get the current position of execution.
func GetPos(skip int) int {
    _, _, line, _ := runtime.Caller(skip + 1)
    return line
}
```
## what I expected

both tests passed, regardless of -race option
## what I saw instead

"go test" passed but "go test -race" failed, because of inconsistent line numbers returned by runtime.Caller.

**Comment by darasion** [comment]
same problem with go1.8, is there any way to detect the `-race` option in the code?

**Comment by ianlancetaylor** [comment]
Actually, this bug is fixed in the upcoming 1.9 release.  I'm going to close this.

It's still broken in 1.8, though, and we won't fix it there.  The only way to detect the -race option in code is to  use a build tag.
